client/GameBoard.jsx

```diff
@@ -280,28 +281,35 @@ export class InnerGameBoard extends React.Component {
     }
 
     getPlots(thisPlayer, otherPlayer) {
+        let commonProps = {
+            cardSize: this.props.user.settings.cardSize,
+            onCardClick: this.onCardClick,
+            onCardMouseOut: this.onMouseOut,
+            onCardMouseOver: this.onMouseOver,
+            onDragDrop: this.onDragDrop,
+            onMenuItemClick: this.onMenuItemClick
+        };
         return (<div className='plots-pane'>
-            <div className='plot-group'>
-                { this.getSchemePile(otherPlayer, false) }
-                <CardPile className={ otherPlayer.plotSelected ? 'plot plot-selected' : 'plot' }
-                    title='Plots' source='plot deck' cards={ otherPlayer.cardPiles.plotDeck }
-                    topCard={ { facedown: true, kneeled: true } } orientation='horizontal'
-                    onMouseOver={ this.onMouseOver } onMouseOut={ this.onMouseOut } disableMouseOver disablePopup
-                    onCardClick={ this.onCardClick } orientation='horizontal' size={ this.props.user.settings.cardSize } />
-                <CardPile className='plot' title='Used Plots' source='revealed plots' cards={ otherPlayer.cardPiles.plotDiscard }
-                    topCard={ otherPlayer.activePlot } orientation='horizontal' onMouseOver={ this.onMouseOver }
-                    onMouseOut={ this.onMouseOut } onCardClick={ this.onCardClick } size={ this.props.user.settings.cardSize } />
-            </div>
-            <div className='plot-group our-side'>
-                <CardPile className='plot' title='Used Plots' source='revealed plots' cards={ thisPlayer.cardPiles.plotDiscard } topCard={ thisPlayer.activePlot }
-                    onMouseOver={ this.onMouseOver } onMouseOut={ this.onMouseOut } orientation='horizontal' onMenuItemClick={ this.onMenuItemClick }
-                    onCardClick={ this.onCardClick } onDragDrop={ this.onDragDrop } size={ this.props.user.settings.cardSize } />
-                <CardPile className={ thisPlayer.plotSelected ? 'plot plot-selected' : 'plot' }
-                    title='Plots' source='plot deck' cards={ thisPlayer.cardPiles.plotDeck } topCard={ { facedown: true, kneeled: true } } orientation='horizontal'
-                    onMouseOver={ this.onMouseOver } onMouseOut={ this.onMouseOut } onCardClick={ this.onCardClick } onDragDrop={ this.onDragDrop }
-                    closeOnClick size={ this.props.user.settings.cardSize } />
-                { this.getSchemePile(thisPlayer, !this.state.spectating) }
-            </div>
+            <PlayerPlots
+                { ...commonProps }
```

Comment:
Let me know if you absolutely hate this syntax and I can drop the corresponding commit. It seemed like a good idea to help emphasize what's different between the two PlayerPlots components (i.e. mostly the player).

If you like it, I should probably go back and apply it to the CardPile rendering in PlayerPlots itself. There'd be more benefit there (3x mostly identical components) than there is here.

Reply:
I think it's fine, but going forward I probably want to encapsulate more of these things in an object that can either be saved/sent via redux or passed around whole so that it doesnt require a zillion attributes to every component.  This will do for now, though